modules/tinymce/src/core/main/ts/api/html/SaxParser.ts

```diff
@@ -118,40 +118,70 @@ const isInvalidUri = (settings: SaxParserSettings, uri: string, tagName: string)
   }
 };
 
+interface LightweightTag {
+  readonly tagDepth: number;
+  readonly end: number;
+}
+
+// This finds a tag, but with very small amounts of detail - used only in findMatchingEndTagIndex
+// Instead of returning what type of tag it is, we just say "is it an open tag, a close tag, or a self-closing tag (like a br, or comment)"
+// And we also don't bother returning the start index of the tag, just the end
+const findLightweightTag = (html: string, startIndex: number, shortEndedElements: SchemaMap): LightweightTag | null => {
+  const startTagRegExp = /<([!?\/])?([A-Za-z0-9\-_:.]+)/;
+  const endTagRegExp = /^(?:\s(?:[^'">]+(?:"[^"]*"|'[^']*'))*[^"'>]*(?:"[^">]*|'[^'>]*)?|\s*|\/)>/;
+
+  while (true) {
+    const startMatch = startTagRegExp.exec(html.slice(startIndex));
+    if (startMatch === null) {
+      return null;
+    }
+    const endOfStart = startIndex + startMatch.index + startMatch[0].length;
+
+    if (startMatch[0] === '<!--') { // it's a comment
```

Comment:
This won't work for a number of cases... it needs to be `<!` to handle bogus comments. From there we need to see if `--` follows and if it does, then it's not bogus, however if that is missing then it is bogus.